include/bluetooth/gatt.h

```diff
@@ -181,6 +181,21 @@ struct bt_gatt_include {
 	uint16_t end_handle;
 };
 
+/** @brief GATT callback structure. */
+struct bt_gatt_cb {
+	/** @brief The ATT MTU has been updated.
+	 *
+	 *  This callback notifies the application that a remote device
+	 *  has changed the ATT MTU.
+	 *
+	 *  @param conn Connection object.
+	 *  @param mtu Updated ATT MTU.
+	 */
+	void (*att_mtu_updated)(struct bt_conn *conn, uint16_t mtu);
```

Comment:
This won't work when there are EATT channels stabilished, since there you can have different MTU for tx/rx and also different sizes between each channe, or you intend to pass only the highest of the channels?

Reply:
I wasn't aware of EATT so there was no intention at all. Are ATT and EATT separate enough that it would make sense to have
```
    void (*att_mtu_updated)(struct bt_conn *conn, uint16_t mtu);
    void (*eatt_mtu_updated)(struct bt_conn *conn, uint8_t channel, uint16_t tx_mtu, uint16_t rx_mtu);
```
Or is there some special eatt reference to signify the old att setup that could be used? Channel = 0 or something?

Reply:
@JordanYates  Seeing as we only have `bt_att_get_mtu` and it returns the highest MTU of all the channel it seems most correct to me to only have one callback, and call it each time the highest possible MTU has increased.

The EATT channels are L2CAP channels and they don't have the MTU exchange procedure, instead the MTU is set based on the L2CAP MTU on the channel.
So in order to call the callback callback for this we would have to use the L2CAP connected callback on the ATT channel.

Alternatively we could document that this is only called on the Fixed ATT channel (or Unenhanced) as a simplification.

Reply:
> @JordanYates Seeing as we only have `bt_att_get_mtu` and it returns the highest MTU of all the channel it seems most correct to me to only have one callback, and call it each time the highest possible MTU has increased.

Agreed with this, @JordanYates could you please implement it this way?

Reply:
> Agreed with this, @JordanYates could you please implement it this way?

I believe the updates implement what you want.

To pick up the EATT channels code is run in `bt_att_connected`, which in turn means the callback is run (due to the fixed ATT channel) before the standard `bt_conn` `connected` callback is run. If this is not desired I  can move that code into an EATT conditional.

The maximum MTU detection code doesn't detect when the MTU has dropped. My understanding is that this could only happen if an EATT channel is closed without the connection dropping. Is this possible?